### DISCUSSION TITLE
Does using PSNR in the conversion process affect the quality?

### DISCUSSION THREAD

**Discussion by mr-qansuh** [staxrip/staxrip#557] [Q&A]
Does using PSNR in the conversion process affect the quality?

**Comment by JJKylee** [comment]
Google it first, please.

It's well documented in `x265` online help [here](https://x265.readthedocs.io/en/master/presets.html?highlight=PSNR#tuning).

And to add a side note:

Quality is very _**subjective**_. Although there are some indexes such as PSNR, SSIM, VMAF, etc. that measure how close the encoded output is to the source, it's the viewer that determines how well the encode is done. Proximity to the source is a totally different story than perceived quality.

**Reply by mr-qansuh** [reply]
i will do google first but i asked about x264

**Reply by JJKylee** [reply]
> i will do google first but i asked about x264

It's in the `x264` console help.

```
- psnr (psy tuning):
--aq-mode 0 --no-psy
```

**Reply by ghost** [reply]
I already answered to  @mr-qansuh a few days ago in another thread.
psnr is a tool for benchmarking. It makes the codec function in a way that the output can be rated and compared to other codecs.
It is not the best way to use the codec.
When you encode, you get better results WITHOUT psnr, except if you want to compare different codecs together according to the psnr standard.

**Reply by mr-qansuh** [reply]
But when i encode without psnr hardsub the subtitle has noise but when i use psnr the quality decrease but hardsub subtitle hasn't noise

**Reply by mr-qansuh** [reply]
I try to encode hardsub without any mistake in subtitle 
So if you have the  Solution 
Notice i would use crf 22 and perset medium

**Reply by JJKylee** [reply]
Like I said before, provide **_proof_**, not just an anecdotal statement.
Your source, encoding settings, logs, and final output.
You’re just requesting, without giving anything from your end.

And if you wanted to talk about it, why didn’t you ask about it in the first place?
Your question was very vague and off the mark.
Please be clear.

**Comment by mr-qansuh** [comment]
this is proof 

video without psnr 
https://sama-share.com/embed-vxrgag5u9rjr-630x430.html

video witht psnr 

https://sama-share.com/embed-a99bfhalfscg-630x430.html

**Please focus on translating**

**Reply by Dendraspis** [reply]
`404 - Not Found`

**Reply by ghost** [reply]
I am downloading to see.

@mr-qansuh preset medium is not good

if you use x264, you must user SLOWER
if you use x265, you must use SLOW

NEVER use medium.

PSNR forced optimizations so the video looks good. When using Medium, some optimizations are disabled in normal mode.
Your file with psnr is larger, you have almost 10% more bit rate. 

I told you and I repeat: PSNR and SSIM are tunes that are made for quality testing.
For normal usage, meaning NOT TESTING, you must not use PSNR (or SSIM) you must use no tune. You must change the other parameters: here, in your case, you must set correctly preset as I explained to you. NO ONE uses x264 with medium preset since 2010.

If you want to keep testing, select in your x264 preset SLOWER (or even SLOW is good) one without psnr and one with psnr, and tell me if you see a difference.

You must also learn that your files include the subtitles stream + the hardcoded subs. So either you include the subtitles stream, or you hardcode it, but not both.

**Reply by ghost** [reply]
@Dendraspis @mr-qansuh 
I find it very rude that this user, @mr-qansuh once he gets an answer, doesn't reply. Then a few days later, he opens a new thread....
He doesn't even thank or acknowledge.

Next time I'll do as @JJKylee suggests in all his answers, and you were right to do that : "google it". 😁

**Reply by Dendraspis** [reply]
@44vince44 
Or simply keep it like I do: don't answer questions, that are not specific in any way or doesn't match a specific level.  😁

**Reply by mr-qansuh** [reply]
> @Dendraspis @mr-qansuh
> I find it very rude that this user, @mr-qansuh once he gets an answer, doesn't reply. Then a few days later, he opens a new thread....
> He doesn't even thank or acknowledge.
> 
> Next time I'll do as @JJKylee suggests in all his answers, and you were right to do that : "google it". 😁

Thank you very much

**Reply by ghost** [reply]
Ah.. that's better now !
